### ISSUE TITLE
Request for Improved Release Documentation

### ISSUE THREAD

**Issue by awesor** [aerospike/aerospike-client-csharp#100]
I was wondering if it would be possible to get a greater amount of detail in the release documentation for major versions.  

In the latest release for example, we seem to have added new code for Cluster Connection Management, yet there isn't anything that mentions why this was added or what initiated the change.

Additionally, we seem to have a new proxy feature for GRPC, but there's not a ton of documentation on how to use it, what advantages it has, etc.  

Is this something we could help with?  Unfortunately in this particular release version, I'm hesitant to upgrade without more insight into what has changed since it's a singular commit with over 13k lines changed.

**Comment by shannonklaus** [comment]
Thanks for your question. We have gotten this feedback to improve the release documentation in the past, and I have been working on improving it. The 7.0.0 release was a big one with the addition of the proxy client. This code is to support our new Aerospike Cloud product and that code shouldn't have an effect on our existing Aerospike users, unless they are interested in making the switch to using the Cloud service. Here is the link to the documentation that I wrote regarding using the new AerospikeClientFactory: https://aerospike.com/docs/cloud/connect/csharp-client
Also note that if you do decide to upgrade to 7.X, the 7.0.0 NuGet package has errors with the dependencies as our release process needed to be improved. 7.0.1 works properly.
Let me know if you have any more questions. I will add additional detail in future release notes, especially for major versions.

**Comment by awesor** [comment]
Thanks @shannonklaus !  

Out of curiosity then just for a quick help on the 7.x upgrade, is there any other major highlight that you'd want to note other than the proxy client?

**Comment by shannonklaus** [comment]
The other more major thing in 7.x is a change to the default of how boolean bins are stored in the server. Previously they were stored as integers and now they will be stored as booleans. I wrote documentation about it here in our [Incompatible API Change page](https://aerospike.com/developer/client/incompatible?client=csharp#version-700), which is updated with every release when relevant.
Other than that, I already alluded to dependency changes in 7.x. We were able to get rid of some and replace with Microsoft code instead, but there is an addition of the libraries needed for the proxy client to work. There were also some minor bug fixes.
